fix: temporary workaround for thorchain and cosmos ledger app check missing in hdwallet

woodenfurniture opened the pull request:
## Description

When attempting to manage accounts for thorchain or cosmos, the app crashes due to missing support for `validateCurrentApp` on these chains. When calling `validateCurrentApp`, the promise is rejected with `Unable to find associated app name for coin: ${coin}` because support for these chains is actually missing from hdwallet.

This issue never presented in web previously because we never attempt to validate the correct app being open on ledger, and instead swallow errors (hence the requirement for the new `throwOnReject` flag added in this PR. 

The workaround here is to bypass this check and allow the user to attempt to proceed, then display an error toast if account fetching fails.

We'll have to follow up with a fix in hdwallet and remove this temporary patch.

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

<!-----------------------------------------------------------------------------
If applicable, please link to the github issue and put `closes #XXXX` in your comment to auto-close the issue that your PR fixes.
------------------------------------------------------------------------------>

progresses #6806  - will not be closed until hdwallet is updated to support these chains and the patch removed from web.

## Risk
> High Risk PRs Require 2 approvals

Low risk

> What protocols, transaction types or contract interactions might be affected by this PR?

## Testing

Attempt to manage thorchain or cosmos accounts without the corresponding ledger app being open on the device. A toast should appear with an appropriate error message.

Managing an account with the corresponding ledger app open should not show error toast, but should behave normally.

### Engineering


### Operations

<!-----------------------------------------------------------------------------
If your changes have a user-facing impact, describe how a non-technical QA team can functionally test your changes in a preview environment.

If they are not user-facing please describe how to test for any regressions that may occur.
------------------------------------------------------------------------------>

## Screenshots (if applicable)


https://github.com/shapeshift/web/assets/125113430/e7f40b23-607b-445f-8a55-c85bd6f41f21

woodenfurniture committed:
fix: temporary workaround for thorchain and cosmos ledger app checks missing in hdwallet

gomesalexandre commented on src/lib/account/account.ts:
```diff
@@ -51,14 +52,19 @@ export const deriveAccountIdsAndMetadata: DeriveAccountIdsAndMetadata = async ar
     (acc, result) => {
       if (isRejected(result)) {
-        console.error(result.reason)
+        if (throwOnReject) {
```
🧠

gomesalexandre commented on src/assets/translations/en/main.json:
```diff
@@ -1811,6 +1811,12 @@
+    },
+    "errors": {
+      "accountMetadataFailedToFetch": {
+        "title": "Account Metadata Fetch Failed",
```
This seems like quite a low-level messaging, but given the root cause is eventually going to get fixed, fine with that!

gomesalexandre reviewed (commented):
Tested locally, confirmed this is happy, though with an issue related to one of the two hard things in software engineering.

Not sure if we want to get this in in this state though however - if this is eventually going to get fixed upstream, it may not worth the effort to fix the caching bug and rather fix the root cause in hdwallet instead? 

- Confirmed it's impossible to currently continue with the wrong app on Ledger ✅ 

<img width="1726" alt="image" src="https://github.com/shapeshift/web/assets/17035424/c2f91ef8-2fdc-468a-be90-9ae34e4a8429">

There's a double toast, although this is eventually going to be removed

- The toast(s) only appear once 🚫 

https://github.com/shapeshift/web/assets/17035424/ee448834-e694-4ac9-8b74-9db92e324c47

- Connecting the correct app is still happy ❓ 

Though on a second try, similarly to the toasts that never appear again, caching means auto-fetching is borked. If hard refreshing the app and trying with the correct app on the first try, this is happy.

https://github.com/shapeshift/web/assets/17035424/6c6aa28a-71ee-4a02-80a7-070f89e29d8c

woodenfurniture committed:
fix: double toast on error

woodenfurniture committed:
fix: stale state on ledger wallets

woodenfurniture commented:
> Tested locally, confirmed this is happy, though with an issue related to one of the two hard things in software engineering.
> 
> Not sure if we want to get this in in this state though however - if this is eventually going to get fixed upstream, it may not worth the effort to fix the caching bug and rather fix the root cause in hdwallet instead?
> 
> * Confirmed it's impossible to currently continue with the wrong app on Ledger ✅
> 
> <img alt="image" width="1726" src="https://private-user-images.githubusercontent.com/17035424/328879881-c2f91ef8-2fdc-468a-be90-9ae34e4a8429.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUyMDIwODYsIm5iZiI6MTcxNTIwMTc4NiwicGF0aCI6Ii8xNzAzNTQyNC8zMjg4Nzk4ODEtYzJmOTFlZjgtMmZkYy00NjhhLWJlOTAtOWFlMzRlNGE4NDI5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTA4VDIwNTYyNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTY2YjNmNjlmYjEzZDc0OGNjYTM2ODM0NGIwYWYyMTFhOTI3YTFhMzMwMjUxMWUzYzA5MTczZTBhNGE2ZmY2N2EmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ABgQuPoE2xv3si7_BRayoX4D77tQL-lW6gzqzLUnugY">
> There's a double toast, although this is eventually going to be removed
> 
> * The toast(s) only appear once 🚫
> 
>  Screen.Recording.2024-05-08.at.14.02.48.mov 
> * Connecting the correct app is still happy ❓
> 
> Though on a second try, similarly to the toasts that never appear again, caching means auto-fetching is borked. If hard refreshing the app and trying with the correct app on the first try, this is happy.
> 
>  Screen.Recording.2024-05-08.at.14.04.03.mov

Ok, thanks for the testing ser! New changes pushed:
* Clear account paging when opening account import for ledger wallets. This fixes the case where it's possible to view accounts from a previous load without the correct ledger app open on the device.
* Fixed double toast on error.
* Moved toast to top-right for consistency with other toasts in account management.

woodenfurniture commented:
Root cause fixed in hdwallet, can remove the workaround but need to keep paging reset.

gomesalexandre commented:
Superseeded by https://github.com/shapeshift/web/pull/6866 with pagination hunks applied there
